src/robot/libdocpkg/writer.py

```diff
@@ -17,14 +17,17 @@
 
 from .htmlwriter import LibdocHtmlWriter
 from .xmlwriter import LibdocXmlWriter
+from .jsonwriter import LibdocJsonWriter
 
 
 def LibdocWriter(format=None, spec_doc_format=None):
     format = (format or 'HTML')
     if format == 'HTML':
-        return LibdocHtmlWriter()
+        return LibdocHtmlWriter(spec_doc_format or 'HTML')
     if format == 'XML':
         return LibdocXmlWriter(spec_doc_format or 'RAW')
     if format == 'LIBSPEC':
         return LibdocXmlWriter(spec_doc_format or 'HTML')
+    if format == 'JSON':
+        return LibdocJsonWriter(spec_doc_format or 'HTML')
```

Comment:
As I commented already with `LibdocJsonWriter`, it would be better to convert docs to HTML if needed already on this side instead of passing `spec_doc_format` to all writers. Could be that doc conversion would be even better done before even calling this factory method.

Reply:
i think that refactoring is out of scope of that issue.
can be done in a later task.

But still i think the best place for that would be the model.

```python
class LibdocJsonWriter(object):

    def __init__(self, spec_doc_format):
        self._spec_doc_format = spec_doc_format

    def write(self, libdoc, outfile):
        libdoc.convert_doc_to(self._spec_doc_format)
        with file_writer(outfile) as f:
            json.dump(libdoc.to_dictionary(), f, indent=2)
```